format/Schema.fbs

```diff
@@ -175,8 +183,39 @@ table Timestamp {
 }
 
 enum IntervalUnit: short { YEAR_MONTH, DAY_TIME}
+
+
+/// A type representing an absolute time difference, also called "timedelta" in
+/// some systems.
+///
+/// SQL systems support many varieties of interval types. The PostgreSQL
+/// interval type is a 16-byte type with microsecond resolution. Microsoft SQL
+/// Server has yet different interval type representations. Some other
+/// libraries, like NumPy, have a 64-bit integer-based timedelta type with time
+/// resolution metadata
+///
+/// We support two styles of intervals encoded in a 64-bit integer
```

Comment:
based on prior threads I'm not clear if both types are 64-bit or only DAY_TIME with YEAR-MONTh being 32-bit.

Reply:
32-bits for YEAR_MONTH intervals is fine with me. The main purpose of this patch is to support other interval resolutions than milliseconds in DAY_TIME

Reply:
If we have two different widths would it make sense to make to separate interval tables?  
It might make the contract a little bit less confusing (YEAR_MONTH has a fixed resolution), DAY_TIME has a configurable one.

Reply:
It looks like this change is only adding a new type, so it think the work here is to just create the the two types.  Then the existing java implementation never becomes part of the IPC spec and should be deprecated (and the 2 new types are officially supported once implemented)?

Reply:
Sorry for deleting last comment, I missed the one white line among all the green :(

@wesm do you agree with the approach?  Do you have bandwidth to update the PR?

Reply:
I can update this PR. I'd like to get @jacques-n or others to sign off on the format changes before it gets merged. 

I don't think the current iteration of the metadata is implemented for IPC in Java, so this is not that disruptive.

Reply:
Actually looking at this file it seems like there are other types that have different widths based on metadata (i.e.date in seconds since epoch are 32 bit and in milliseconds 64 bits, so we can probably keep as is).

Still ramping up on the java side of things, but it looks like the easiest path forward to not break existing class/structure hierarchy or cause undue confusion is to add a new IntervalUnit (maybe called EPOCH?) above and denote DAY_TIME as "optional" in the standard, but still have it reflect the current java definition (days and milliseconds stored contiguosly as 2 int32s, https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/IntervalDayVector.java)